sdcm/utils/k8s/chaos_mesh.py

```diff
@@ -0,0 +1,172 @@
+            use_devel=False,
+            namespace=self.NAMESPACE,
+            values=HelmValues(self.HELM_SETTINGS | {
+                "chaosDaemon": scylla_node_pool_affinity,
+                "controllerManager": aux_node_pool_affinity,
+                "dnsServer": aux_node_pool_affinity
+            }),
+            atomic=True,
+            timeout="30m"
+        )
+        LOGGER.info("chaos-mesh installed successfully on %s k8s cluster.", self._k8s_cluster.k8s_scylla_cluster_name)
+
+
+class ExperimentStatus(Enum):
+    STARTING = 0
+    RUNNING = 1
+    PAUSED = 2
+    FINISHED = 3
+    ERROR = 4
+    UNKNOWN = 5
+
+
+class PodChaosExperiment:
+    """Base class for all PodChaos experiments."""
+    API_VERSION = "chaos-mesh.org/v1alpha1"
+
+    def __init__(self, pod: "BasePodContainer", name: str):
+        self._k8s_cluster = pod.parent_cluster.k8s_cluster
+        self._name = name
+        self._namespace = pod.parent_cluster.namespace
+        self._experiment = {
+            "apiVersion": self.API_VERSION,
+            "metadata": {
+                "name": self._name,
+                "namespace": self._namespace
+            }
+        }
+
+    def start(self):
+        """Starts experiment. Does not wait for finish."""
+        LOGGER.debug("Starting a pod-failure experiment %s", self._name)
+        assert self._k8s_cluster, "K8s cluster hasn't been configured for this experiment."
+        with NamedTemporaryFile(suffix=".yaml", mode="w") as experiment_config_file:
+            yaml.dump(self._experiment, experiment_config_file)
+            experiment_config_file.flush()
+            self._k8s_cluster.apply_file(experiment_config_file.name)
+        LOGGER.info("pod-failure experiment '%s' has started", self._name)
+
+    def get_status(self) -> ExperimentStatus:
+        """Gets status of podchaos experiment."""
+        result = self._k8s_cluster.kubectl(
+            f"get podchaos {self._name}  -n {self._namespace} -o jsonpath='{{.status.conditions}}'", verbose=False)
+        condition = {cond["type"]: ast.literal_eval(cond["status"]) for cond in json.loads(result.stdout)}
+        if not condition["Selected"] and condition["Paused"]:
+            return ExperimentStatus.ERROR
+        if not condition["Selected"] and not condition["Paused"] and not condition["AllRecovered"] and not condition["AllInjected"]:
+            return ExperimentStatus.STARTING
+        if condition["Selected"] and not condition["Paused"] and not condition["AllRecovered"] and condition["AllInjected"]:
+            return ExperimentStatus.RUNNING
+        if condition["AllRecovered"] and condition["Paused"]:
+            return ExperimentStatus.PAUSED
+        if condition["AllRecovered"] and not condition["Paused"]:
+            return ExperimentStatus.FINISHED
+        LOGGER.warning("Unknown experiment status: %s", condition)
+        return ExperimentStatus.UNKNOWN
+
+    def wait_until_finished(self, timeout: int):
```

Comment:
in `PodFailureExperiment` we specify the duration, it would be nice if the code could use the information from the experiment to figure out what's the timeout should be. 

and not force who write it to keep track, and do the calculation of how much extra time it's gonna give this function.